drivers/adc/adc_stm32.c

```diff
@@ -1526,15 +1480,16 @@ static int adc_stm32_init(const struct device *dev)
 
 	config->irq_cfg_func();
 
-#if defined(CONFIG_SOC_SERIES_STM32F1X) || \
-	defined(STM32F3X_ADC_V2_5)
-	/*
-	 * Calibration of F1 and F3 (ADC1_V2_5) must starts after two cycles
-	 * after ADON is set.
-	 */
-	LL_ADC_StartCalibration(adc);
+#if defined(CONFIG_SOC_SERIES_STM32F1X) || defined(STM32F3X_ADC_V2_5)
+	wait_cycles = SystemCoreClock / adc_rate *
+		      STM32F1_ADC_DELAY_CALIB_ENABLE_ADC_CYCLES;
```

Comment:
could adc_rate  be null ?

Reply:
It clearly shouldn't be, but in doubt, I added a check. Also moved this waiting in a specific function.